Library/Homebrew/cask/lib/hbc/cask.rb

```diff
@@ -85,6 +85,15 @@ def installed?
       !versions.empty?
     end
 
+    def latest_installed_version
+      versions.last
+    end
+
+    def outdated?(greedy: false)
+      (greedy || !auto_updates && !version == :latest) &&
+      !version.nil? && (version.casecmp latest_installed_version) == 1
```

Comment:
Version is nil when you've installed casks that aren't yet in a tap. E.g I have a couple of casks that I wrote that I submitted PRs for but never followed up to complete (sorry). They show up with a ! in brew cask list. This check is here as the call to casecmp fails otherwise. We need the casecmp for a later than check rather than an equality check.